### DISCUSSION TITLE
Yazilan Kodlarda Paket Tanimi Yapilmasi

### DISCUSSION THREAD

**Discussion by mesutckr** [gusanmaz/BMB212-Algorithms-2223#16] [Q&A] [answered]
Merhaba,
Soruları windows üzerinde eclipse kullanarak yaptım. her soru için klasör içerisinde Student.java ve main.java dosyaları oldu. Fakat bunlar içerisinde student.java da dahil en üst satırda çalışma klasörü paket olarak eklemek zorunda kaldım. sisteme yüklerken bu dosyaları nasıl yüklemem gerekir? Paket tanımlaması yapmak gerekir mi?

**Comment by gusanmaz** [comment, marked as answer]
Paket tanimi yapmaktan kacinalim kodlarda. Java paketlerini hic duymamis biri de bu odevleri kodlayabilir.

Projenizdeki tum Java dosyalari kok dizinde olursa (diger bir deyisle projenizde herhangi bir klasor olusturmazsaniz) Java dosyalari icin herhangi bir paket tanimi yapma ihtiyacinin olmamasi lazim.

Her bir odev icin o odevin davet linki kullanilarak bir repo olusturulmasi gerekiyor. Olusturulan repoda sadece o odevle ilgili dosyalarin olmasi gerekiyor. 

Verilen ilk 6 odevde tum dosyalar kok dizininde bulunmali. Diger bir deyisle reponuzda herhangi bir klasor olusturmayin.

**Reply by guidesoftware** [reply]
Student.java dosyasını student adlı klasöre atmadığım zaman main.java dosyası import edilen student.java dosyasını görmüyor ve okumuyor. bu yüzden student klasörü oluşturup içine student.javayı atmak zorunda kalıyorum. bu şekilde import ediliyor. visual studio code kullanırken böyle bir sorunla karşılaşıyorum. bu şekil göndersem , sorun olur mu ?

**Reply by gusanmaz** [reply]
`Student.java` ve `Main.java` beraber kok klasorde olursa sorun cikmamasi lazim. Dosyalari kok dizinine yerlestirip VS Code terminal ekranindan `javac` komutu ile derlemeyi denediniz mi? 

Ayrica IntelliJ IDEA'da `Student.java` ve `Main.java` dosyalarini ayni satira koydugumda projeyi derleme ve calistirma konusunda bir sorun yasamadim.

Otamatik not degerlendirme sistemi repolarinizi ayri ayri klasorlere indirip her klasor icinde `javac` komutunu kullanarak kodunuzu derleyip, test edecek.
